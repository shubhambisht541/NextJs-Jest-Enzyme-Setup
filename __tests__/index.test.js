import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import App from '../pages/index.js'
import MessageList from '../components/messageList'

describe('When the index page is loaded', () => {
	const initialState = {}
	const mockStore = configureStore()
	let wrapper, store;

	beforeEach(() => {
		store = mockStore(initialState)
    wrapper = shallow(<App store={store} />).dive();
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render one <MessageList> component', () => {
    expect(wrapper.find(MessageList)).toHaveLength(1);
  })

  it('Should render one <input> element', () => {
    expect(wrapper.find('input')).toHaveLength(1);
  })

  it('Should render one <button> element', () => {
    expect(wrapper.find('button')).toHaveLength(1);
  })
})


describe('When the input text is empty', () => {
	const initialState = {}
	const mockStore = configureStore()
	let wrapper, store;

	beforeEach(() => {
		store = mockStore(initialState)
    wrapper = shallow(<App store={store} />).dive();
  })

  it('Should <button> element is disabled', () => {
    expect(wrapper.find('input').props().value).toBe("");
    expect(wrapper.find('button').props().disabled).toBe(true);
  })
})


describe('When the input text is filled', () => {
	const initialState = {}
	const mockStore = configureStore()
	let wrapper, store;

	beforeEach(() => {
		store = mockStore(initialState)
    wrapper = shallow(<App store={store} />).dive();
  })

  it('Should <button> element is enabled', () => {
    wrapper.find('input').simulate('change', { target: { value: 'text' }});
    expect(wrapper.find('button').props().disabled).toBe(false);
  })

  it('Should update the state.message', () => {
    wrapper.find('input').simulate('change', { target: { value: 'text' }});
    expect(wrapper.state('message')).toEqual('text');
  })
})
