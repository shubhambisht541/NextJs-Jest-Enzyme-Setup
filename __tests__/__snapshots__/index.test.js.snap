// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the index page is loaded renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(MessageList) />,
        <form
          className="message-form"
          onSubmit={[Function]}
        >
          <input
            onChange={[Function]}
            placeholder="Type your message"
            type="text"
            value=""
          />
          <button
            disabled={true}
            type="submit"
          >
            Send
          </button>
        </form>,
      ],
      "className": "chat",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Type your message"
              type="text"
              value=""
            />,
            <button
              disabled={true}
              type="submit"
            >
              Send
            </button>,
          ],
          "className": "message-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Type your message",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Send",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Send",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(MessageList) />,
          <form
            className="message-form"
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Type your message"
              type="text"
              value=""
            />
            <button
              disabled={true}
              type="submit"
            >
              Send
            </button>
          </form>,
        ],
        "className": "chat",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                placeholder="Type your message"
                type="text"
                value=""
              />,
              <button
                disabled={true}
                type="submit"
              >
                Send
              </button>,
            ],
            "className": "message-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Type your message",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Send",
                "disabled": true,
                "type": "submit",
              },
              "ref": null,
              "rendered": "Send",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
